packages/browser-ppx/ppx.ml

```diff
@@ -273,14 +276,202 @@ module Browser_only = struct
+    match stri.pstr_desc with
+    | Pstr_typext { ptyext_attributes = attrs; _ }
+    | Pstr_modtype { pmtd_attributes = attrs; _ }
+    | Pstr_open { popen_attributes = attrs; _ }
+    | Pstr_include { pincl_attributes = attrs; _ }
+    | Pstr_exception { ptyexn_attributes = attrs; _ }
+    | Pstr_primitive { pval_attributes = attrs; _ }
+    | Pstr_eval (_, attrs)
+    | Pstr_module { pmb_attributes = attrs; _ } ->
+        if should_keep attrs = `keep then Some stri else None
+    | Pstr_value (_, vbs) ->
+        if should_keep_many vbs (fun vb -> vb.pvb_attributes) = `keep then
+          Some stri
+        else None
+    | Pstr_type (recflag, tds) ->
+        if should_keep_many tds (fun td -> td.ptype_attributes) = `keep then
+          let tds = apply_config_on_types tds in
+          Some { stri with pstr_desc = Pstr_type (recflag, tds) }
+        else None
+    | Pstr_recmodule md ->
+        if should_keep_many md (fun md -> md.pmb_attributes) = `keep then
+          Some stri
+        else None
+    | Pstr_class cds ->
+        if should_keep_many cds (fun cd -> cd.pci_attributes) = `keep then
+          Some stri
+        else None
+    | Pstr_class_type ctds ->
+        if should_keep_many ctds (fun ctd -> ctd.pci_attributes) = `keep then
+          Some stri
+        else None
+    | Pstr_extension _ | Pstr_attribute _ -> Some stri
+
+  let apply_config_on_signature_item sigi =
+    match sigi.psig_desc with
+    | Psig_typext { ptyext_attributes = attrs; _ }
+    | Psig_modtype { pmtd_attributes = attrs; _ }
+    | Psig_open { popen_attributes = attrs; _ }
+    | Psig_include { pincl_attributes = attrs; _ }
+    | Psig_exception { ptyexn_attributes = attrs; _ }
+    | Psig_value { pval_attributes = attrs; _ }
+    | Psig_modtypesubst { pmtd_attributes = attrs; _ }
+    | Psig_modsubst { pms_attributes = attrs; _ }
+    | Psig_module { pmd_attributes = attrs; _ } ->
+        if should_keep attrs = `keep then Some sigi else None
+    | Psig_typesubst tds ->
+        if should_keep_many tds (fun td -> td.ptype_attributes) = `keep then
+          let tds = apply_config_on_types tds in
+          Some { sigi with psig_desc = Psig_typesubst tds }
+        else None
+    | Psig_type (recflag, tds) ->
+        if should_keep_many tds (fun td -> td.ptype_attributes) = `keep then
+          let tds = apply_config_on_types tds in
+          Some { sigi with psig_desc = Psig_type (recflag, tds) }
+        else None
+    | Psig_recmodule md ->
+        if should_keep_many md (fun md -> md.pmd_attributes) = `keep then
+          Some sigi
+        else None
+    | Psig_class cds ->
+        if should_keep_many cds (fun cd -> cd.pci_attributes) = `keep then
+          Some sigi
+        else None
+    | Psig_class_type ctds ->
+        if should_keep_many ctds (fun ctd -> ctd.pci_attributes) = `keep then
+          Some sigi
+        else None
+    | Psig_extension _ | Psig_attribute _ -> Some sigi
+
+  let preprocess_impl _exp_ctxt str =
```

Comment:
I am not sure what the `_exp_ctxt` variable is about, it was introduced in ppxlib v2, @davesnx do you know?